### PR TITLE
Document how the dirty __eq__ is called

### LINKED ISSUE #38: Include documentation on how this works
I find this functionality quite interesting and think it could be useful.
But I'm a bit hesitant to include a library with "magical" behavior that I don't understand.

My main question is, why is the `__eq__` method of the right operand used?
The [python documentation](https://docs.python.org/3/reference/datamodel.html#object.__eq__) seems to suggest that `x == y` should call `x.__eq__(y)`.

>  If the operands are of different types, and right operand’s type is a direct or indirect subclass of the left operand’s type, the reflected method of the right operand has priority, otherwise the left operand’s method has priority.

The last paragraph in the docs above point me to the `__eq__` implementation in the `DirtyEqualsMeta` class. But I'm not sure what is going on.

### PR DESCRIPTION
**Pull request by Marco-Kaulea**
Closes #38

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,3 +97,13 @@ pip install dirty-equals
 ```
 
 **dirty-equals** requires **Python 3.7+**.
+
+## Internals
+### How is is the magical `DirtyEquals.__eq__` called?
+When you call `x == y`, Python first calls `x.__eq__(y)`. This would not help us much, because we would have to keep an eye on order of the arguments when comparing to `DirtyEquals` objects. But that's where were another feature of Python comes in.
+
+When `x.__eq__(y)` returns the `NotImplemented` object, then Python will try to call `y.__eq__(x)`. Objects in the standard library return that value when they don't know how to compare themselves to objects of `type(y)` (Without checking the C source I can't be certain if this assumption holds for all classes, but it works for all the basic ones). In [pathlib.PurePath](https://github.com/python/cpython/blob/aebbd7579a421208f48dd6884b67dbd3278b71ad/Lib/pathlib.py#L751) you can see an example how that is implemented in Python.
```
Comment (samuelcolvin) [inline README.md:105]:
```suggestion
When `x.__eq__(y)` returns the `NotImplemented` object, then Python will try to call `y.__eq__(x)`. Objects in the standard library return that value when they don't know how to compare themselves to objects of `type(y)` (Without checking the C source I can't be certain if this assumption holds for all classes, but it works for all the basic ones). In [`pathlib.PurePath`](https://github.com/python/cpython/blob/aebbd7579a421208f48dd6884b67dbd3278b71ad/Lib/pathlib.py#L751) you can see an example how that is implemented in Python.
```

File: README.md [hunk 2]
```diff
@@ -97,3 +97,13 @@ pip install dirty-equals
...
+
+> By default, object implements eq() by using is, returning NotImplemented in the case of a false comparison: True if x is y else NotImplemented.
+
+See the Python documentation for more information ([object.__eq__](https://docs.python.org/3/reference/datamodel.html#object.__eq__])).
```
Comment (samuelcolvin) [inline README.md:109]:
```suggestion
See the Python documentation for more information ([`object.__eq__`](https://docs.python.org/3/reference/datamodel.html#object.__eq__)).
```

File: README.md [hunk 3]
```diff
@@ -97,3 +97,13 @@ pip install dirty-equals
...
 
 **dirty-equals** requires **Python 3.7+**.
+
+## Internals
+### How is is the magical `DirtyEquals.__eq__` called?
+When you call `x == y`, Python first calls `x.__eq__(y)`. This would not help us much, because we would have to keep an eye on order of the arguments when comparing to `DirtyEquals` objects. But that's where were another feature of Python comes in.
+
+When `x.__eq__(y)` returns the `NotImplemented` object, then Python will try to call `y.__eq__(x)`. Objects in the standard library return that value when they don't know how to compare themselves to objects of `type(y)` (Without checking the C source I can't be certain if this assumption holds for all classes, but it works for all the basic ones). In [pathlib.PurePath](https://github.com/python/cpython/blob/aebbd7579a421208f48dd6884b67dbd3278b71ad/Lib/pathlib.py#L751) you can see an example how that is implemented in Python.
+
+> By default, object implements eq() by using is, returning NotImplemented in the case of a false comparison: True if x is y else NotImplemented.
```
Comment (samuelcolvin) [inline README.md:107]:
```suggestion
> By default, object implements `__eq__()` by using `is`, returning `NotImplemented` in the case of a false comparison: `True if x is y else NotImplemented`.
```

**Review by samuelcolvin — COMMENTED** [review]
Thanks so much for investigating this.

Arguably this belongs more in the docs than in the readme, but I'm happy for it to be included in both. Please could you copy or move this into `docs/index.md` or a new file in docs.

Please could you also wrap this content to match the style of the rest of the readme and docs.

**Comment by Marco-Kaulea** [comment]
I wasn't sure where to put it initially, so the bottom of the README seemed as good a place as any. But on second though it's probably not important enough.

I hope the line wrapping is OK this way, that was my editor default.

**Review by samuelcolvin — COMMENTED** [review]
Some more typos :-)

**Comment by samuelcolvin** [comment]
You need to add `internals.md` to `mkdocs.yml`, you can see in [the preview](https://smokeshow.helpmanual.io/180i4h41310o5l205329/) that the file isn't rendered. You might also want to change the heading levels to 1 and 2.

**Comment by samuelcolvin** [comment]
or you can add this content to the bottom of `index.md`, whichever you prefer.

**Comment by Marco-Kaulea** [comment]
No idea where I can find the current preview. I hope this works now.
Either way, I'm off of work now and might not check back in for a while.

**Pull request merged**